### DISCUSSION TITLE
Implementation of Multiple views on GLSP editor

### DISCUSSION THREAD

**Discussion by Abhay0294** [eclipse-glsp/glsp#710] [Q&A]
Hi,

My requirement is as follows[please have a look at the snap for reference]:

I'm having an UI on the GLSP editor itself which consists of 2 buttons(Button A and Button B).[Already implemented]
Button A is currently selected and is having two nodes(Node A & Node B).
If Button B is clicked, it should open another fresh GLSP editor view(**not the GLSP editor instance**) and the user should able to add nodes or any editor related things.**[Requirement]**
If user again moves to the Button A he/she should get those Node A and Node B as previous.
If possible please le me know how this can be achieved.

![GLSP_Issue](https://user-images.githubusercontent.com/57217189/178906453-d4af4efd-4966-47ab-8bfb-de3990375d7a.png)

**Comment by planger** [comment]
Hi @Abhay0294,

from the top of my head I would send a `RequestModelAction` with custom parameters (indicating whether the context is A or B) to the server on click of the button A or B. The server could then react by sending the respective diagrams back in a `SetModelAction`.
As this is not standard behavior there might be some other customization, especially on the server handlers, required to deal with this scenario, but it should be possible to make it work.

So the idea is that the server actually has one model as usual, but knows based on some context, whether to generate a diagram for `A` or `B` in its `GModelFactory`. With the `RequestModelAction` mentioned above, you could switch this context, so that the server would generate and send the correct diagram from the model.

Hope this helps!
